runtime/device/android/wenet/app/src/main/cpp/CMakeLists.txt

```diff
@@ -5,38 +5,33 @@ set(CMAKE_CXX_STANDARD 14)
 
 include_directories(${CMAKE_SOURCE_DIR})
 
-file(GLOB wenet_SOURCES
-    wenet.cpp
-    decoder/ctc_prefix_beam_search.cc
-    decoder/torch_asr_decoder.cc
-    decoder/torch_asr_model.cc
-    frontend/feature_pipeline.cc
-    frontend/fft.cc
-    )
-
-add_library(${TARGET} SHARED
-    ${wenet_SOURCES}
-    )
-
+# Pytorch
 set(build_DIR ${CMAKE_SOURCE_DIR}/../../../build)
 file(GLOB PYTORCH_INCLUDE_DIRS "${build_DIR}/pytorch_android*.aar/headers")
 file(GLOB PYTORCH_LINK_DIRS "${build_DIR}/pytorch_android*.aar/jni/${ANDROID_ABI}")
-
-target_compile_options(${TARGET} PRIVATE
-    -fexceptions
-    )
-
 find_library(PYTORCH_LIBRARY pytorch_jni
     PATHS ${PYTORCH_LINK_DIRS}
-    NO_CMAKE_FIND_ROOT_PATH
-    )
+    NO_CMAKE_FIND_ROOT_PATH)
+find_library(FBJNI_LIBRARY fbjni
+    PATHS ${PYTORCH_LINK_DIRS}
+    NO_CMAKE_FIND_ROOT_PATH)
+include_directories(${PYTORCH_INCLUDE_DIRS}
+    ${PYTORCH_INCLUDE_DIRS}/torch/csrc/api/include)
+
+# frontend
+add_library(frontend STATIC
+    frontend/feature_pipeline.cc
+    frontend/fft.cc)
+target_link_libraries(frontend PUBLIC log)
+
+# decoder
+add_library(decoder STATIC
+    decoder/ctc_prefix_beam_search.cc
+    decoder/torch_asr_decoder.cc
+    decoder/torch_asr_model.cc)
+target_link_libraries(decoder PUBLIC log ${PYTORCH_LIBRARY} ${FBJNI_LIBRARY})
 
-target_include_directories(${TARGET} PRIVATE
-    ${PYTORCH_INCLUDE_DIRS}
-    ${PYTORCH_INCLUDE_DIRS}/torch/csrc/api/include
-    )
+link_libraries(frontend decoder android)
```

Comment:
what is android here?

Reply:
It means link libraries to all following targets.

Reply:
libandroid_jni is needed by libpython_jni.